src/clients/SpokePoolClient.ts

```diff
@@ -835,6 +836,15 @@ export class SpokePoolClient extends BaseAbstractClient {
           ? { ...(spreadEventWithBlockNumber(event) as V3FillWithBlock), destinationChainId: this.chainId }
           : { ...(spreadEventWithBlockNumber(event) as V2FillWithBlock) };
 
+        if (isV3Fill(fill) && Array.isArray(fill.relayExecutionInfo)) {
```

Comment:
ooc, did you manage to confirm that this type is never an array on other chains?

Also, worth an `assert(fill.relayExecutionInfo.length === 4)` ?

Reply:
I found it also on Arbitrum now